src/lib/customHooks.ts

```diff
@@ -0,0 +1,33 @@
+import { usersApi } from '@/api/api';
+import { useState, useEffect } from 'react';
+import { toUser, User } from './user';
+
+export function useUser() {
+  const [user, setUser] = useState<User | null>(null);
+
+  const getAuthenticatedUser = (): Promise<User | null> => {
+    return new Promise((resolve, reject) => {
+      usersApi
+        .usersControllerGetProfile()
+        .then((response) => {
+          resolve(toUser(response));
+        })
+        .catch(() => {
+          reject(null);
+        });
+    });
+  };
+
+  useEffect(() => {
+    async function getUserDetails() {
+      getAuthenticatedUser()
+        .then((user) => {
+          setUser(user);
+        })
+        .catch();
+    }
+    getUserDetails();
+  }, []);
```

Comment:
If we are not logged, we get an error in the console, we shouldn't have that

Reply:
Ok, I'll see what I can do to solve this issue.